ponys.js

```
@@ -9,6 +9,7 @@ export default class {
 
 	static define(name, template, options, url = '')
 	{
+		if (customElements.get(name)) return Promise.reject(Error(`Component '${name}' already registered`));
```

The browser already throws a DOMException if you try to redefine a custom element, which also rejects the Promise returned below - so I'd prefer to just let it happen!

---

While yes that is true, there is SOOO much that happens between the top of that function call and the time that `customElements.define` is called (parsing, encoding, imports, class building, etc), that if we know ahead of time that the call will fail anyways, we should just fail early. Just my two-cents...

---

I agree there's often a good argument for an early return in such cases, and if it was preventing end-users from waiting around on imports that will only lead to failing custom element defs, then yeah, I'd do it... but I think it's highly unlikely that any such redefinition errors will happen in practice given that the responsibility for naming these tags is entirely in the hands of the library-user, and such errors should certainly be caught before they make it to production (assuming the fix for the silent fails in `defineAll()` below! :wink:). So given that, I'd prefer to let the browser throw its `DOMException`, to give the clearest indication to devs what went wrong and where.